### PR TITLE
Bug fix/fix internal issue 4451

### PR DESCRIPTION
**Pull request by hkernbach**
Same as: https://github.com/arangodb/arangodb/pull/10538

### CODE REVIEW

**Comment by hkernbach** [comment]
Jenkins: http://jenkins01:8080/view/PR/job/arangodb-matrix-pr/7382/ (expected to fail)

**Comment by hkernbach** [comment]
http://jenkins01:8080/view/PR/job/arangodb-matrix-pr/7407/

**Comment by hkernbach** [comment]
http://jenkins01:8080/view/PR/job/arangodb-matrix-pr/7485/

File: arangod/GeneralServer/RestHandler.cpp [hunk 1]
```diff
@@ -113,7 +113,20 @@ futures::Future<Result> RestHandler::forwardRequest(bool& forwarded) {
     return futures::makeFuture(Result());
   }
 
-  std::string serverId = forwardingTarget();
+  ResultT forwardResult = forwardingTarget();
+  if (forwardResult.fail()) {
+    return futures::makeFuture(forwardResult.result());
+  }
+
+  auto forwardContent = forwardResult.get();
```
Comment (mchacki) [inline arangod/GeneralServer/RestHandler.cpp]: maybe we can do
```suggestion
  auto const [serverId, removeHeader] = forwardResult.get();
```
Here, but this is not urgent, and will not block this PR

Comment (mchacki) [reply arangod/GeneralServer/RestHandler.cpp]: Do not commit this, from UI.
Needs modification of the lines below.

Comment (hkernbach) [reply arangod/GeneralServer/RestHandler.cpp]: First I've implemented with auto. But a lambda cb had issues with it, Row: 199:
```
 auto cb = [this, serverId, useVst,
             self = shared_from_this()](network::Response&& response) -> Result {
```

**Comment by hkernbach** [comment]
http://jenkins01:8080/view/PR/job/arangodb-matrix-pr/7488/

**Comment by hkernbach** [comment]
http://jenkins01:8080/view/PR/job/arangodb-matrix-pr/7497/

**Comment by hkernbach** [comment]
Jenkins green!

**Comment by KVS85** [comment]
Tests blue.

**Pull request merged**